Library/ViewModels/LiveStreamContainerViewModel.swift

```diff
@@ -0,0 +1,213 @@
+  var creatorAvatarLiveDotImageViewHidden: Signal<Bool, NoError> { get }
+  var creatorIntroText: Signal<String, NoError> { get }
+  var dismiss: Signal<(), NoError> { get }
+  var liveStreamState: Signal<LiveStreamViewControllerState, NoError> { get }
+  var loaderStackViewHidden: Signal<Bool, NoError> { get }
+  var loaderText: Signal<String, NoError> { get }
+  var navBarLiveDotImageViewHidden: Signal<Bool, NoError> { get }
+  var numberWatchingButtonHidden: Signal<Bool, NoError> { get }
+  var projectImageUrl: Signal<URL?, NoError> { get }
+  var showErrorAlert: Signal<String, NoError> { get }
+  var videoViewControllerHidden: Signal<Bool, NoError> { get }
+  var titleViewText: Signal<String, NoError> { get }
+}
+
+public final class LiveStreamContainerViewModel: LiveStreamContainerViewModelType,
+LiveStreamContainerViewModelInputs, LiveStreamContainerViewModelOutputs {
+
+  //swiftlint:disable function_body_length
+  //swiftlint:disable cyclomatic_complexity
+  public init() {
+    let configData = Signal.combineLatest(
+      self.configData.signal.skipNil(),
+      self.viewDidLoadProperty.signal
+      )
+      .map(first)
+
+    let project = configData.map(first)
+
+    let event = Signal.merge(
+      configData.map(second).skipNil(),
+      self.liveStreamEventProperty.signal.skipNil()
+    )
+
+    self.createAndConfigureLiveStreamViewController = Signal.combineLatest(project, event)
+      .take(first: 1)
+      .map { project, event in (project, AppEnvironment.current.currentUser?.id, event) }
+
+    self.liveStreamState = Signal.combineLatest(
+      Signal.merge(
+        self.liveStreamViewControllerStateChangedProperty.signal.skipNil(),
+        project.mapConst(.loading)
+      ),
+      self.viewDidLoadProperty.signal
+    ).map(first)
+
+    self.showErrorAlert = self.liveStreamState.map { state -> LiveVideoPlaybackError? in
+      switch state {
+      case .error(let error): return error
+      case .live(let playbackState, _):
+        if case let .error(videoError) = playbackState { return videoError }
+      case .replay(let playbackState, _):
+        if case let .error(videoError) = playbackState { return videoError }
+      case .initializationFailed:
+        return .failedToConnect
+      default:
+        return nil
+      }
+
+      return nil
+      }
+      .skipNil()
+      .map {
+        switch $0 {
+        case .sessionInterrupted:
+          return Strings.The_live_stream_was_interrupted()
+        case .failedToConnect:
+          return Strings.The_live_stream_failed_to_connect()
+        }
+      }
+
+    self.loaderText = Signal.merge(
+      liveStreamState.map {
+        if case .live(playbackState: .loading, _) = $0 { return Strings.The_live_stream_will_start_soon() }
+        if case .greenRoom = $0 { return Strings.The_live_stream_will_start_soon() }
+        if case .replay(playbackState: .loading, _) = $0 {
+          return Strings.The_replay_will_start_soon()
+        }
```

Comment:
@justinswart since we are `if case`ing this so much it might be better to just do a switch